src/collects/seashell/backend/asan-error-parse.rkt

```diff
@@ -0,0 +1,344 @@
+     (define-values (framelist lines-left) (try-parse-stack-frame lines))
+     (SectionData "free-non-malloced-address" ; error type
+                  (jsexpr `((framelist ,framelist) (misc ,extra-info)))
+                  #f ; global extra info
+                  lines-left))))
+
+;; Try to parse a frame
+(define frame-parse : SectionParser
+  (match-and-process
+   #px"^\\{\"frame\": "
+   (lambda ([error-type : String] [lines : (Listof String)] [match-result : (Listof String)])
+     (define-values (framelist lines-left) (try-parse-stack-frame lines))
+     (SectionData #f ; error type
+                  (jsexpr `((framelist ,framelist) (misc ,(hash))))
+                  #f ; global extra info
+                  lines-left))))
+
+;; A big list of functions to try and parse the input lines.
+(define all-parsers : ParserList
+  (list (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: SEGV on unknown address 0x0+ " "segmentation-fault-on-null-address")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: SEGV" "segmentation-fault")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: stack-buffer-overflow" "stack-buffer-overflow")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: global-buffer-overflow" "global-buffer-overflow")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: heap-buffer-overflow" "heap-buffer-overflow")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: heap-use-after-free"  "heap-use-after-free")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: stack-use-after-return"  "stack-use-after-return")
+        (match-type #px"^=+\\d+=+ERROR: AddressSanitizer: stack-use-after-scope"  "stack-use-after-scope")
+        (match-type #px"^=+\\d+=+ERROR: LeakSanitizer: detected memory leaks" "memory-leak")
+        memory-leak-parser
+        stack-overflow-parser function-info array-parser
+        heap-address-details global-address-details
+        double-free double-free-first-free allocation-details
+        free-non-malloc
+        frame-parse
+        ))
+
+(: try-parsers ((Listof String) String JSList JSHash  ParserList -> SectionData))
+(define (try-parsers lines error-type call-stacks extra-info section-parsers)
+  (cond [(or (empty? lines) (empty? section-parsers)) no-data]
+        [else (define result ((first section-parsers) lines error-type call-stacks extra-info))
+              (if (equal? result no-data)
+                  (try-parsers lines error-type call-stacks extra-info (rest section-parsers))
+                  result)]))
+
+
+;; Purpose: Consumes a string raw-asan-output which is the raw output from ASAN. Will parse raw-asan-output, stripping
+;; out "scary" stuff that students don't care about, and putting important information in a JSON format for the front-end.
+;; This function is mostly just a wrapper that converts the input into a list of strings, passes it to asan-parser,
+;; which does most of the work.
+(: asan->json (Bytes -> Bytes))
```

Comment:
Don't I have to eventually convert the JSExpr to Bytes to send it to the web browser?